pkg/didexchange/invitation/invitation.go

```diff
@@ -0,0 +1,85 @@
+// +build js,wasm
+
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package invitation
+
+import (
+	"encoding/json"
+	"fmt"
+	"syscall/js"
+	"time"
+
+	"github.com/hyperledger/aries-framework-go/pkg/client/didexchange"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+)
+
+// RegisterHandleInvitationJSCallback register handle invitation js callback
+func RegisterHandleInvitationJSCallback(didexchangeClient *didexchange.Client) {
+	statusCh := make(chan service.StateMsg, 10)
+	if err := didexchangeClient.RegisterMsgEvent(statusCh); err != nil {
+		js.Global().Call("alert", err.Error())
+	}
+
+	c := callback{didexchangeClient: didexchangeClient, statusCh: statusCh}
+	js.Global().Set("acceptInvitation", js.FuncOf(c.handleInvitation))
+}
+
+type callback struct {
+	didexchangeClient *didexchange.Client
+	statusCh          chan service.StateMsg
+}
+
+func (c *callback) handleInvitation(this js.Value, inputs []js.Value) interface{} {
+	// https://github.com/golang/go/issues/26382
+	go func() {
+		invitationData := js.Global().Get("document").
+			Call("getElementById", inputs[0].String()).Get("value").String()
+		if invitationData == "" {
+			js.Global().Call("alert", "invitation data is empty")
```

Comment:
We will need to refactor (later) to have UI elements instead of alerts.

Reply:
yes